pkg/sql/parser/scan.go

```diff
@@ -795,14 +812,28 @@ func isNonKeywordBareIdentifier(s string) bool {
 			return false
 		}
 	}
-	upper := strings.ToUpper(s)
-	if _, ok := reservedKeywords[upper]; ok {
+
+	// If we made it this far, we know there's no unicode characters in our
+	// string, so we can use a naive unicode-unaware uppercase algorithm that's
+	// faster than Strings.ToUpper.
+	// See https://github.com/golang/go/issues/17859 for ToUpper performance.
+	isLower := true
+	b := make([]byte, len(s))
```

Comment:
Same thing here, it's too bad we have to allocate even if `s` already is all uppercase.